src/v2i-hub/CARMAStreetsPlugin/src/JsonToJ2735SSMConverter.cpp

```diff
@@ -0,0 +1,156 @@
+#include "JsonToJ2735SSMConverter.h"
+
+namespace CARMAStreetsPlugin
+{
+
+    bool JsonToJ2735SSMConverter::parseJsonString(const string &consumedMsg, Json::Value &ssmDoc) const
+    {
+        const auto jsonLen = static_cast<int>(consumedMsg.length());
+        Json::CharReaderBuilder builder;
+        JSONCPP_STRING err;
+        const std::unique_ptr<Json::CharReader> reader(builder.newCharReader());
+        bool parseResult = reader->parse(consumedMsg.c_str(), consumedMsg.c_str() + jsonLen, &ssmDoc, &err);
+        if (!parseResult)
+        {
+            std::cout << "Parse error: " << err << endl;
+        }
+        return parseResult;
+    }
+
+    void JsonToJ2735SSMConverter::toJ2735SSM(const Json::Value &ssmDoc, std::shared_ptr<SignalStatusMessage> ssmPtr) const
+    {
+        try 
+        {        
+            ASN_STRUCT_FREE_CONTENTS_ONLY(asn_DEF_SignalStatusMessage, ssmPtr.get());
+            if (!ssmDoc.isMember("SignalStatus"))
+            {
+                return;
```

Comment:
Maybe add a log statement to indicate something has gone wrong here

Reply:
https://github.com/usdot-fhwa-OPS/V2X-Hub/pull/533/commits/53ec358d4de07c0a7567a9b5dfe7e86e0a5624c3